File: src/jmh/java/ru/sbt/ss/quicksort/benchmark/ParallelQuicksortBenchmark.java
package ru.sbt.ss.quicksort.benchmark;

import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.Param;
import ru.sbt.ss.quicksort.ParallelQuickSort;

public class ParallelQuicksortBenchmark extends QuickSortBenchmark {
    @Param(value = {"1", "2", "4", "8"})
    int parallelism = 0;
    @Benchmark
    public void parallelUsingForkJoinPool() {
        ParallelQuickSort.parallelQuickSortUsingForkJoinPool(arr, parallelism);
    }
}

File: src/jmh/kotlin/ru/sbt/ss/quicksort/ParallelQuickSortBenchmark.kt
package ru.sbt.ss.quicksort

import org.openjdk.jmh.annotations.Benchmark
import org.openjdk.jmh.annotations.Param
import ru.sbt.ss.quicksort.sort.parallelQuickSortUsingCoroutines

@Suppress("unused")
open class ParallelQuickSortBenchmark : QuickSortBenchmark() {
    @Param(value = ["4"])
    var parallelism: Int = 0

    @Benchmark
    fun parallelUsingCoroutines() {
        array.parallelQuickSortUsingCoroutines(parallelism)
    }
}